src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolution.cs

```diff
@@ -3034,9 +3036,14 @@ private RefKind GetEffectiveParameterRefKind(ParameterSymbol parameter, RefKind
             //   exists from the argument to the type of the corresponding parameter, or
             // - for a ref or out parameter, the type of the argument is identical to the type of the corresponding parameter. 
 
-            // RefKind has to match unless the ref kind is None and argument expression is of the type dynamic. This is a bug in Dev11 which we also implement. 
+            // There are two cases where ref kind can mismatch:
+            // 1) If the ref kind is None and argument expression is of the type dynamic. This is a bug in Dev11 which we also implement. 
             // The spec is correct, this is not an intended behavior. We don't fix the bug to avoid a breaking change.
-            if (argRefKind != parRefKind && !(argRefKind == RefKind.None && argument.HasDynamicType()))
+            // 2) PROTOTYPE(readonly-ref) If it was an extension method 'this' argument, and the parameter is ref or ref-readonly.
+
+            if (argRefKind != parRefKind &&
+                !(argRefKind == RefKind.None && argument.HasDynamicType()) &&
+                !(isThisArgument && (parRefKind == RefKind.Ref || parRefKind == RefKind.RefReadOnly)))
```

Comment:
>RefReadOnly [](start = 88, length = 11)

do we need to check for RefReadOnly?  I think we represent them as None for the purpose of overload resolution.